### PR TITLE
Validate user name format

### PR DESCRIPTION
**Pull request by djohnston32**
I rule and have completed some work on Case Manager that's ready for review!

Add validation on "name" field for user model.

This pull request makes the following changes:
* Adds regex for validating name format
* Adds a test to ensure we're still allowing valid names with special characters like dashes and accent marks
* Adds a test for notable invalid cases: injecting a website name, accidentally supplying an email address in the name field

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

**Review by colinxfleming — CHANGES_REQUESTED** [review]
I should have pulled this for you earlier, but here's the characters that currently appear:

```
irb(main):014:0> names.join('').split('').uniq.sort.join('')
=> "\t '(),-.@ABCDEFGHIJKLMNOPQRSTUVWYZabcdefghijklmnopqrstuvwxyzáèëó
```

there's only one instance of @ and one instance of a name with a period in it, so I feel comfortable taking those out. There are a BUNCH of parens though so we'll probably want to allowlist those? No idea about the tabs though, ha.

So I think the two actions here are:
* add an error message to tell people how to fix scrub names
* loosen the req here a little bit to allow quotes, parens, and commas?

then I think this is good to go. thanks for this, it's great work.

**Comment by djohnston32** [comment]
Hmm, we may have a bit of a problem if there's a user with a period in their name. That one's really important to block since that's the only way to avoid e.g. "evil.com" without explicitly listing out domain patterns to block. But leaving the blanket validation against the period means that existing user won't be able to update their name or password. Best option might be to allow "untouched" name fields somehow so we can blanket allow names that already exist. Still causes problems if somebody named "firstname.lastname" decides to change to "newfirstname.newlastname" but maybe that's ok.

**Comment by djohnston32** [comment]
I just threw an `if: :name_changed?` on there and that seems to work.

File: test/controllers/users_controller_test.rb [hunk 1]
```diff
@@ -120,6 +120,28 @@ class UsersControllerTest < ActionDispatch::IntegrationTest
...
+        attributes[:name] = name
+        assert_difference 'User.count', 1 do
+          post users_path, params: { user: attributes }
+        end
+        assert_equal name, User.last.name
+      end
+    end
+
+    it 'should fail with error message for invalid name' do
+      ['wikipedia.org', '*****', 'asdf@email.com'].each do |name|
```
[the remark this answers is not in the document]
Comment (djohnston32) [reply test/controllers/users_controller_test.rb:136]: No problem!

**Comment by djohnston32** [comment]
> So I think the two actions here are:
> 
>     * add an error message to tell people how to fix scrub names
> 
>     * loosen the req here a little bit to allow quotes, parens, and commas?

It's hard to tell but the quote-looking thing in the regex is a backtick not an apostrophe, so quotes, parens, and commas are all allowed with the current regex. I included a name with an apostrophe in the tests to be extra sure it's allowed since I figure that's something especially easy to get wrong.

**Comment by djohnston32** [comment]
Just realized since parens are allowed, I don't need the change removing parens in the admin user name. Will fix and updated.

File: app/models/user.rb [hunk 2]
```diff
@@ -43,6 +43,12 @@ class User < ApplicationRecord
   # email presence validated through Devise
   validates_uniqueness_to_tenant :email, allow_blank: true, if: :email_changed?
   validates :name, :role, :email, presence: true
+  validates :name,
+            format: {
+              with: /\A[^0-9`!@#\$%\^&*+_=.\/]*\z/,
+              message: "may not include numbers or the following symbols: `!@#$%^&*+_=./"
+            },
+            if: :name_changed?
```
Comment (colinxfleming) [inline app/models/user.rb:51]: `name_changed` is a great way of handling this. great call here

**Review by colinxfleming — APPROVED** [review]
really great work here. thank you very much for thinking through this problem so thoroughly!

**Pull request merged**